### PR TITLE
JIT: enhance SSA checker to check some PHI properties

### PR DESCRIPTION
**Pull request by AndyAyersMS**
* Ensure PhiArgs have the right local number.
* Ensure PhiArgs have unique `gtPredBB` (for most blocks)

Handler entries may have multiple PhiArgs with the same `gtPredBB`. This is by design, to model exceptional flow from the middle of the block to a handler.

Jump threading relies on there being just a single PhiArg per pred. So exclude handler entry blocks from jump threading.

Prep work for possibly changing SSA to produce one PhiArg per pred, instead of one PhiArg per ssa def.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@jakobbotsch PTAL
cc @dotnet/jit-contrib 

No diffs expected.

File: src/coreclr/jit/fgdiagnostic.cpp [hunk 1]
```diff
@@ -4200,6 +4200,76 @@ class SsaCheckVisitor : public GenTreeVisitor<SsaCheckVisitor>
...
+    }
+
+    void CheckPhis(BasicBlock* block)
+    {
+        for (Statement* const stmt : block->Statements())
+        {
+            if (!stmt->IsPhiDefnStmt())
+            {
+                break;
+            }
```
Comment (BruceForstall) [inline src/coreclr/jit/fgdiagnostic.cpp:4213]: We already have NonPhiStatements(); maybe we should have PhiStatements()? (Presumably somebody else would verify all the PHI statements appear first).

```suggestion
        for (Statement* const stmt : block->PhiStatements())
        {
```

**Review by BruceForstall — APPROVED** [review]

**Comment by AndyAyersMS** [comment]
Failure looks similar to #85082 but the pattern there is different.

```
CSC : error CS8034: Unable to load Analyzer assembly /__w/1/s/.dotnet/sdk/8.0.100-preview.3.23178.7/Sdks/Microsoft.NET.Sdk/analyzers/Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll : The file '/tmp/VBCSCompiler/AnalyzerAssemblyLoader/a6fe1c15dfaa4493b9e3eda990245fe8/30' already exists. [/__w/1/s/src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.csproj] [/__w/1/s/src/tests/build.proj]
```

File: src/coreclr/jit/fgdiagnostic.cpp (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/fgdiagnostic.cpp:4213]: I should probably check that all the PhiDefs appear first, but then I wouldn't need the partial iterator.

**Pull request merged**